sdk-springboot/src/main/java/io/dapr/springboot/DaprBeanPostProcessor.java

```diff
@@ -80,27 +85,25 @@ private static void subscribeToTopics(Class clazz, EmbeddedValueResolver embedde
         continue;
       }
 
-      String route = topic.name();
-      PostMapping mapping = method.getAnnotation(PostMapping.class);
-
-      if (mapping != null && mapping.path() != null && mapping.path().length >= 1) {
-        route = mapping.path()[0];
-      } else if (mapping != null && mapping.value() != null && mapping.value().length >= 1) {
-        route = mapping.value()[0];
-      }
-
       String topicName = embeddedValueResolver.resolveStringValue(topic.name());
       String pubSubName = embeddedValueResolver.resolveStringValue(topic.pubsubName());
       if ((topicName != null) && (topicName.length() > 0) && pubSubName != null && pubSubName.length() > 0) {
         try {
           TypeReference<HashMap<String, String>> typeRef
                   = new TypeReference<HashMap<String, String>>() {};
           Map<String, String> metadata = MAPPER.readValue(topic.metadata(), typeRef);
-          DaprRuntime.getInstance().addSubscribedTopic(pubSubName, topicName, route, metadata);
+          List<String> routes = SpringProcessorHelper.getAllCompleteRoutesForPost(clazz, method, topic.name());
```

Comment:
```suggestion
          List<String> routes = SpringProcessorHelper.getAllCompleteRoutesForPost(clazz, method, topicName);
```
Always use the resolved name.

Reply:
Agreed.